qiskit_experiments/characterization/qubit_spectroscopy.py

```diff
@@ -45,203 +48,62 @@ class SpectroscopyAnalysis(BaseAnalysis):
-                used for Kerneled data and a conversion from counts to probabilities will
-                be done if Discriminated data was measured.
-            amp_guess: The amplitude of the Gaussian function, i.e. :math:`a`. If not
-                provided, this will default to -1 or 1 depending on the measured values.
-            sigma_guesses: The guesses for the standard deviation of the Gaussian distribution.
-                If it is not given this will default to an array of ten
-                points linearly spaced between zero and width of the x-data.
-            freq_guess: A guess for the frequency of the peak :math:`x0`. If not provided
-                this guess will default to the location of the highest or lowest point of
-                the y-data depending on the y-data.
-            offset_guess: A guess for the magnitude :math:`b` offset of the fit function.
-                If not provided, the initial guess defaults to the median of the y-data.
-            amp_bounds: Bounds on the amplitude of the Gaussian function as a tuple of
-                two floats. The default bounds are (-1, 1).
-            sigma_bounds: Bounds on the standard deviation of the Gaussian function as a tuple
-                of two floats. The default values are (0, frequency range).
-            freq_bounds: Bounds on the center frequency as a tuple of two floats. The default
-                values are (min(frequencies) - df, max(frequencies) - df).
-            offset_bounds: Bounds on the offset of the Gaussian function as a tuple of two floats.
-                The default values are (-2, 2).
-            plot: If True generate a plot of fitted data.
-            ax: Optional, matplotlib axis to add the plot to.
-            kwargs: Trailing unused function parameters.
-
-        Returns:
-            The analysis result with the estimated peak frequency and the plots if a plot was
-            generated.
-
-        Raises:
-            QiskitError:
-                - If the measurement level is not supported.
-                - If the fit fails.
-        """
-
-        meas_level = experiment_data.data(0)["metadata"]["meas_level"]
-        meas_return = experiment_data.data(0)["metadata"]["meas_return"]
-
-        # Pick a data processor.
-        if data_processor is None:
-            data_processor = get_to_signal_processor(meas_level=meas_level, meas_return=meas_return)
-            data_processor.train(experiment_data.data())
-
-        y_sigmas = np.array([data_processor(datum) for datum in experiment_data.data()])
-        min_y, max_y = min(y_sigmas[:, 0]), max(y_sigmas[:, 0])
-        ydata = (y_sigmas[:, 0] - min_y) / (max_y - min_y)
-
-        # Sigmas may be None and fitting will not work if any sigmas are exactly 0.
-        try:
-            sigmas = y_sigmas[:, 1] / (max_y - min_y)
-            if any(sigmas == 0.0):
-                sigmas = None
```

Comment:
It's supported in the fitter rather than here. 
https://github.com/Qiskit/qiskit-experiments/blob/0729d6cb6e4349f77f9b2ba7b5af75df72dbd6be/qiskit_experiments/analysis/curve_fitting.py#L129-L135

Because this is not only the issue of spectroscopy, but also causes zero div error in other experiments.

Reply:
Makes sense. Thanks.